microceph/ceph/rbd_mirror.go

```diff
@@ -464,6 +486,60 @@ func configureImageFeatures(pool string, image string, op string, feature string
 	return nil
 }
 
+// enableImageFeatures enables the list of rbd features on the requested resource.
+func enableRbdImageFeatures(poolName string, imageName string, features []string) error {
+	for _, feature := range features {
+		err := configureImageFeatures(poolName, imageName, "enable", feature)
+		if err != nil && !strings.Contains(err.Error(), "one or more requested features are already enabled") {
+			return err
+		}
+	}
+	return nil
+}
+
+// disableRbdImageFeatures disables the list of rbd features on the requested resource.
+func disableRbdImageFeatures(poolName string, imageName string, features []string) error {
+	for _, feature := range features {
+		err := configureImageFeatures(poolName, imageName, "disable", feature)
+		if err != nil {
```

Comment:
I wonder if this should be fatal, would it be worthwhile to only a log a WARN and try to carry on?

Reply:
sometimes, we may try to disable a feature which was already disabled by a user or external agent. This can lead to failures for all ops affected.